src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -176,7 +176,7 @@ def build_nsg_resource(_, name, location, tags, nsg_rule_type):
     nsg = {
         'type': 'Microsoft.Network/networkSecurityGroups',
         'name': name,
-        'apiVersion': '2015-06-15',
+        'apiVersion': '2017-09-01' if support_asg else '2015-06-15',
```

Comment:
I've become less and less convinced that putting a safe minimum API here makes sense. I'm increasingly growing to believe we should just bite the bullet and put the actual  Networking API version here. If you grab me we can discuss.